src/components/SearchBlock/SearchBlock.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Select } from 'antd';
+import { changeSearch, resetSearch } from '../../modules/search';
+import './antd-select.css';
+import './style.css';
+
+// Path for img when I use server static file
+// src='./img/logo.png'
+
+const { Option } = Select;
+const initialState = {
+  isKeyWasDown: false,
+  searchQuery: '',
+  searchLocation: 'Location',
+};
+
+class SearchBlock extends Component {
+  constructor(props) {
+    super();
+    this.state = initialState;
+    this.handleChangeInput = ({ target: { name, value } }) => {
+      this.setState({ [name]: value.trim(), isKeyWasDown: !!value });
+    };
+    this.handleChangeLocation = searchLocation => this.setState({ searchLocation, isKeyWasDown: true });
+    this.handleClickSearch = (e) => {
+      e.preventDefault();
+      const { searchQuery, searchLocation } = this.state;
+      const { changeSearch } = this.props;
+      if (searchQuery || searchLocation !== 'Location') {
+        const searchQueryProp = searchQuery || null;
+        const searchLocationProp = searchLocation !== 'Location' ? searchLocation : null;
+        changeSearch({ searchQuery: searchQueryProp, searchLocation: searchLocationProp });
+      }
+    };
+    this.handleClickReset = (e) => {
+      const { resetSearch } = this.props;
+      e.preventDefault();
+      resetSearch();
+      this.setState(initialState);
+    };
+  }
+
+  render() {
+    const { searchQuery, searchLocation, isKeyWasDown } = this.state;
```

Comment:
isKeyWasDown is a weird construction even in terms of English language. Choose one time option: isKeyDown or wasKeyDown. Don't make English speaking devs cry

Reply:
ahah, okay